nft_ingester/src/program_transformers/bubblegum/collection_verification.rs

```diff
@@ -1,75 +1,111 @@
-                            asset_id: Set(id_bytes.clone()),
-                            group_key: Set("collection".to_string()),
-                            group_value: Set(Some(collection.to_string())),
-                            seq: Set(seq as i64),
-                            slot_updated: Set(bundle.slot as i64),
-                            ..Default::default()
-                        };
-                        let mut query = asset_grouping::Entity::insert(grouping)
-                            .on_conflict(
-                                OnConflict::columns([
-                                    asset_grouping::Column::AssetId,
-                                    asset_grouping::Column::GroupKey,
-                                ])
-                                .update_columns([
-                                    asset_grouping::Column::GroupKey,
-                                    asset_grouping::Column::GroupValue,
-                                    asset_grouping::Column::Seq,
-                                    asset_grouping::Column::SlotUpdated,
-                                ])
-                                .to_owned(),
-                            )
-                            .build(DbBackend::Postgres);
-                        query.sql = format!(
+        let asset_to_update = asset::ActiveModel {
+            id: Unchanged(id_bytes.clone()),
+            leaf: Set(Some(le.leaf_hash.to_vec())),
+            seq: Set(seq as i64),
+            data_hash: Set(Some(updated_data_hash.clone())), // todo remove clone
+            ..Default::default()
+        };
+        update_compressed_asset(txn, id_bytes.clone(), Some(seq), asset_to_update).await?;
+
+        if verify {
+            let grouping = asset_grouping::ActiveModel {
+                asset_id: Set(id_bytes.clone()),
+                group_key: Set("collection".to_string()),
+                group_value: Set(Some(collection.to_string())),
+                seq: Set(seq as i64),
+                slot_updated: Set(bundle.slot as i64),
+                ..Default::default()
+            };
+            let mut query = asset_grouping::Entity::insert(grouping)
+                .on_conflict(
+                    OnConflict::columns([
+                        asset_grouping::Column::AssetId,
+                        asset_grouping::Column::GroupKey,
+                    ])
+                    .update_columns([
+                        asset_grouping::Column::GroupKey,
+                        asset_grouping::Column::GroupValue,
+                        asset_grouping::Column::Seq,
+                        asset_grouping::Column::SlotUpdated,
+                    ])
+                    .to_owned(),
+                )
+                .build(DbBackend::Postgres);
+            query.sql = format!(
                     "{} WHERE excluded.slot_updated > asset_grouping.slot_updated AND excluded.seq >= asset_grouping.seq",
```

Comment:
Yeah not sure if we want the slot_updated here because there could be multiple tx in the same slot when just dealing with bubblegum transactions?